smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -264,4 +273,83 @@ private String getSanitizedMemberName(MemberShape member) {
     private boolean isRequiredMember(MemberShape member) {
         return member.isRequired() && !member.hasTrait(IdempotencyTokenTrait.class);
     }
+
+    /**
+     * Writes a const map of member validators into the namespace for use by the validate method.
+     *
+     * @param writer the writer for the type, currently positioned in the type's exported namespace
+     */
+    void writeMemberValidators(TypeScriptWriter writer) {
+        writer.openBlock("const memberValidators = {", "};", () -> {
+            for (MemberShape member : members) {
+                Collection<Trait> constraintTraits = getConstraintTraits(member);
+                if (constraintTraits.isEmpty()) {
+                    writer.addImport("NullValidator", "__NullValidator", "@aws-smithy/server-common");
+                    writer.write("$L: new __NullValidator(),", getSanitizedMemberName(member));
+                } else {
+                    writer.addImport("CompositeValidator", "__CompositeValidator", "@aws-smithy/server-common");
+                    writer.openBlock("$L: new __CompositeValidator<$T>([", "]),",
+                            getSanitizedMemberName(member),
+                            symbolProvider.toSymbol(member),
+                            () -> {
+                        for (Trait t : constraintTraits) {
+                            writeValidator(writer, t);
+                        }
+                    });
+                }
+            }
+        });
+    }
+
+    void writeValidate(TypeScriptWriter writer, String param) {
+        writer.openBlock("return [", "];", () -> {
+            for (MemberShape member : members) {
+                writer.write("...memberValidators.$1L.validate($2L.$1L, $3S),",
+                        getSanitizedMemberName(member), param, member.getMemberName());
+            }
+        });
+    }
+
+    private void writeValidator(TypeScriptWriter writer, Trait trait) {
+        if (trait instanceof RequiredTrait) {
+            writer.addImport("RequiredValidator", "__RequiredValidator", "@aws-smithy/server-common");
```

Comment:
`@aws-smithy/server-common` is repeated in many places. I'm not sure, but wonder if these should be specified in AwsDependency/TypeScriptDependency. I've seen others defined there.